src/drawers/drawer.ts

```diff
@@ -5,238 +5,130 @@
-        selection.attr(colorAttribute, step.attrToAppliedProjector[colorAttribute]);
-      }
-    });
-    step.animator.animate(selection, step.attrToAppliedProjector);
-    if (this._className != null) {
-      this.selection().classed(this._className, true);
-    }
-  }
-
-  protected _drawStepCanvas(data: any[], step: Drawers.AppliedDrawStep) {
-    Utils.Window.warn("canvas rendering not yet implemented on " + (<any> this.constructor).name);
-  }
-
-  private _appliedProjectors(attrToProjector: AttributeToProjector): AttributeToAppliedProjector {
-    const modifiedAttrToProjector: AttributeToAppliedProjector = {};
-    Object.keys(attrToProjector).forEach((attr: string) => {
-      modifiedAttrToProjector[attr] =
-        (datum: any, index: number) => attrToProjector[attr](datum, index, this._dataset);
-    });
-
-    return modifiedAttrToProjector;
+  public getCanvas(): d3.Selection<HTMLCanvasElement, any, any, any> {
+    return this._canvas;
   }
 
   /**
-   * Calculates the total time it takes to use the input drawSteps to draw the input data
-   *
-   * @param {any[]} data The data that would have been drawn
-   * @param {Drawers.DrawStep[]} drawSteps The DrawSteps to use
-   * @returns {number} The total time it takes to draw
+   * Removes this Drawer's renderArea
    */
-  public totalDrawTime(data: any[], drawSteps: Drawers.DrawStep[]) {
-    let delay = 0;
-    drawSteps.forEach((drawStep, i) => {
-      delay += drawStep.animator.totalTime(data.length);
-    });
-
-    return delay;
+  public remove() {
+    if (this._svgRootG != null) {
+      this._svgRootG.remove();
+    }
   }
 
   /**
-   * Draws the data into the renderArea using the spefic steps and metadata
+   * Draws the data using the drawSteps onto the currently selected drawer.
    *
    * @param{any[]} data The data to be drawn
    * @param{DrawStep[]} drawSteps The list of steps, which needs to be drawn
    */
-  public draw(data: any[], drawSteps: Drawers.DrawStep[]) {
-    const appliedDrawSteps: Drawers.AppliedDrawStep[] = drawSteps.map((dr: Drawers.DrawStep) => {
-      const attrToAppliedProjector = this._appliedProjectors(dr.attrToProjector);
+  public draw(data: any[], drawSteps: DrawStep[]) {
+    const appliedDrawSteps: AppliedDrawStep[] = drawSteps.map((drawStep) => {
+      const attrToAppliedProjector = applyProjectorsWithDataset(drawStep.attrToProjector, this._dataset);
       return {
         attrToAppliedProjector: attrToAppliedProjector,
-        animator: dr.animator,
+        animator: drawStep.animator,
       };
     });
 
-    if (this._renderArea != null) {
-      this._bindSelectionData(data);
-      this._cachedSelectionValid = false;
-
-      let delay = 0;
-      appliedDrawSteps.forEach((drawStep, i) => {
-        Utils.Window.setTimeout(() => this._drawStep(drawStep), delay);
-        delay += drawStep.animator.totalTime(data.length);
-      });
+    if (this._svgRootG != null) {
```

Comment:
maybe this should null check the _svgDrawer and the one below can check _canvasDraw